advanced_functionality/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,743 @@
+    "  1. [When should I build my own algorithm container?](#When-should-I-build-my-own-algorithm-container?)\n",
+    "  1. [Permissions](#Permissions)\n",
+    "  1. [The example](#The-example)\n",
+    "  1. [The presentation](#The-presentation)\n",
+    "1. [Part 1: Packaging and Uploading your Algorithm for use with Amazon SageMaker](#Part-1:-Packaging-and-Uploading-your-Algorithm-for-use-with-Amazon-SageMaker)\n",
+    "    1. [An overview of Docker](#An-overview-of-Docker)\n",
+    "    1. [How Amazon SageMaker runs your Docker container](#How-Amazon-SageMaker-runs-your-Docker-container)\n",
+    "      1. [Running your container during training](#Running-your-container-during-training)\n",
+    "        1. [The input](#The-input)\n",
+    "        1. [The output](#The-output)\n",
+    "      1. [Running your container during hosting](#Running-your-container-during-hosting)\n",
+    "    1. [The parts of the sample container](#The-parts-of-the-sample-container)\n",
+    "    1. [The Dockerfile](#The-Dockerfile)\n",
+    "    1. [Building and registering the container](#Building-and-registering-the-container)\n",
+    "  1. [Testing your algorithm on your local machine](#Testing-your-algorithm-on-your-local-machine)\n",
+    "1. [Part 2: Training and Hosting your Algorithm in Amazon SageMaker](#Part-2:-Training-and-Hosting-your-Algorithm-in-Amazon-SageMaker)\n",
+    "  1. [Set up the environment](#Set-up-the-environment)\n",
+    "  1. [Create the session](#Create-the-session)\n",
+    "  1. [Upload the data for training](#Upload-the-data-for-training)\n",
+    "  1. [Training On SageMaker](#Training-on-SageMaker)\n",
+    "  1. [Making Predictions using Boto3](#Making-predictions-using-Boto3)\n",
+    "  1. [Optional cleanup](#Optional-cleanup)  \n",
+    "1. [Reference](#Reference)\n",
+    "\n",
+    "_or_ I'm impatient, just [let me see the code](#The-Dockerfile)!\n",
+    "\n",
+    "## When should I build my own algorithm container?\n",
+    "\n",
+    "You may not need to create a container to bring your own code to Amazon SageMaker. When you are using a framework such as Apache MXNet or TensorFlow that has direct support in SageMaker, you can simply supply the Python code that implements your algorithm using the SDK entry points for that framework. This set of supported frameworks is regularly added to, so you should check the current list to determine whether your algorithm is written in one of these common machine learning environments.\n",
+    "\n",
+    "Even if there is direct SDK support for your environment or framework, you may find it more effective to build your own container. If the code that implements your algorithm is quite complex or you need special additions to the framework, building your own container may be the right choice.\n",
+    "\n",
+    "This walkthrough shows that it is quite straightforward to build your own container. So don't worry if there isn't direct SDK support for your environment.\n",
```

Comment:
I'd change to: "So you can still use SageMaker even if your use case is not covered by the deep learning containers that we've built for you."